tests/testthat/test-write_csv_file.R

```diff
@@ -0,0 +1,7 @@
+test_that("with readr < 1.4.0 and named argument `file` throws no error", {
+  skip_if(utils::packageVersion("readr") >= "1.4.0")
+
+  x <- tibble::tibble(x = 1)
+  file <- tempfile()
+  expect_error(write_csv_file(x = x, file = file), NA)
+})
```

Comment:
Should we test the converse of this too, that there's no warning if we have `readr` >= 1.4.0? That was the primary intention of all this.

Reply:
That warning is was a "good" warning, reminding the users of `path` that they should use `file`. That is a behavior of `readr::write_csv()` that I would preserve. Also It wouldn't be super easy to implement, I think, as `write_csv_file()` passes arguments to `readr::write_csv()` via the elipsis `...`, so to intercept `path` we would need to see what arguments the user passed via `...`, then rename `path` as `file`. It seems like a lot of room for error to suppress a behavior that is safe, and that will eventually go away -- maybe years from now -- whenever `readr::write_csv()` decides to forget about `path`.

In the interest of moving forward I'll merge, assuming that my interpretation of the issue is correct, and that my resistance to implement it makes sense. But do feel free to follow up and we can open another PR.

Reply:
Could we use `expect_silent()` for testing the other side? https://testthat.r-lib.org/reference/expect_silent.html

Reply:
Oh, I now think you mean to see if the caller of the new `write_csv_file()` effectively throws no warning?

export_audit_invalid_data
![image](https://user-images.githubusercontent.com/5856545/114182498-86bfe880-98f7-11eb-9f93-048910e4ad27.png)

Maybe I could figure out a way to mock the inputs, but I would be okay just looking at the output in Ci should be fine

https://github.com/2DegreesInvesting/PACTA_analysis/pull/439/checks?check_run_id=2298956928

![image](https://user-images.githubusercontent.com/5856545/114183016-29786700-98f8-11eb-8928-295695352b32.png)

Reply:
Here I see the relevant warning went away.

Reply:
you're right... I'm still wrapping my head around the consequences of this

I think my idea was, we started this thread to prevent these warnings from showing up in the log under normal, working conditions... so that's the primary goal. But yeah, I guess we have for sure achieved that by always using the proper argument name depending on the `readr` version.